tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+from subprocess import check_output
+from sys import argv, exit
+from tests.configurable_cluster import ConfigurableCluster
+from tests.docker_cluster import DockerCluster, DockerClusterException
+from tests.no_hadoop_bare_image_provider import NoHadoopBareImageProvider
+
+from tests.product.mode_installers import StandaloneModeInstaller, \
+    YarnSliderModeInstaller
+from tests.product.prestoadmin_installer import PrestoadminInstaller
+from tests.product.standalone.presto_installer import StandalonePrestoInstaller
+from tests.product.topology_installer import TopologyInstaller
+from tests.product.yarn_slider.slider_installer import SliderInstaller
+
+import argparse
+
+
+class ProductTestImageBuilder:
+    STANDALONE_BARE_CLUSTER = 'bare'
+    BARE_CLUSTER = 'bare'
+    PA_ONLY_CLUSTER = 'pa_only_standalone'
+    STANDALONE_PRESTO_CLUSTER = 'presto'
+
+    PA_ONLY_YS_CLUSTER = 'pa_only_ys'
+    PA_SLIDER_CLUSTER = 'pa_slider'
+
+    _cluster_types = {
+        BARE_CLUSTER: [],
+        PA_ONLY_CLUSTER: [PrestoadminInstaller,
+                          StandaloneModeInstaller],
+        STANDALONE_PRESTO_CLUSTER: [PrestoadminInstaller,
+                                    StandaloneModeInstaller,
+                                    TopologyInstaller,
+                                    StandalonePrestoInstaller],
+        PA_ONLY_YS_CLUSTER: [PrestoadminInstaller,
+                             YarnSliderModeInstaller],
+        PA_SLIDER_CLUSTER: [PrestoadminInstaller,
+                            YarnSliderModeInstaller,
+                            SliderInstaller]
+    }
+
+    def __init__(self):
+        self.maxDiff = None
+        self.cluster = None
+        self.default_keywords = {}
```

Comment:
Each installer type has some keywords that it can access. For instance, the presto installer will have:

`{
        'rpm': self.rpm_name,
        'rpm_basename': RPM_BASENAME
}`